src/test/ui/rust-2018/dyn-keyword.rs

```diff
@@ -0,0 +1,10 @@
+// edition:2015
+// run-rustfix
+
+#![allow(unused_variables)]
+#![deny(keyword_idents)]
+
+fn main() {
+    let dyn = (); //~ ERROR dyn
+    //~^ WARN hard error in the 2018 edition
+}
```

Comment:
+ You'll want tests that this is correctly rustfix-ed.
+ Also, a test that this isn't a proper keyword on Rust 2015 is needed.

Reply:
I forgot to commit the `.stderr` and `.fixed` files 🙄
I think this test should cover not being a proper keyword on Rust 2015, as the error message specifically pertains to the 2018 edition.